cypress/integration/console/gateways/general-settings.spec.js

```diff
@@ -0,0 +1,123 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Gateway general settings', () => {
+  let user
+  let gateway
+
+  before(() => {
+    cy.dropAndSeedDatabase()
+    user = {
+      ids: { user_id: 'gtw-settings-test-user' },
+      primary_email_address: 'gtw-settings-test-user@example.com',
+      password: 'ABCDefg123!',
+      password_confirm: 'ABCDefg123!',
+    }
+    cy.createUser(user)
+    gateway = {
+      ids: { gateway_id: 'test-gateway', eui: '0000000000000000' },
+      name: 'Test Gateway',
+      description: 'Gateway for testing gateway general settings',
+      schedule_anytime_delay: '523ms',
+      enforce_duty_cycle: true,
+      gateway_server_address: 'localhost',
+    }
+  })
+
+  it('displays gateway values that are set', () => {
+    cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+    cy.createGateway(gateway, user.ids.user_id)
+    cy.visit(
+      `${Cypress.config('consoleRootPath')}/gateways/${gateway.ids.gateway_id}/general-settings`,
+    )
+    cy.findByText('General settings', { selector: 'h1' }).should('be.visible')
+    cy.findByText('Basic settings', { selector: 'h3' }).should('be.visible')
+    cy.findByText('General settings, gateway updates and metadata', { selector: 'p' }).should(
+      'be.visible',
+    )
+    cy.findByText('General settings', { selector: 'h4' }).should('be.visible')
+    cy.findByLabelText('Gateway ID')
+      .should('be.disabled')
+      .and('have.attr', 'value')
+      .and('eq', gateway.ids.gateway_id)
+    cy.findByLabelText('Gateway EUI')
+      .should('not.be.disabled')
+      .and('have.attr', 'value')
+      .and('eq', gateway.ids.eui)
+    cy.findByLabelText('Gateway Name')
+      .should('be.visible')
+      .and('have.attr', 'value')
+      .and('eq', gateway.name)
+    cy.findByLabelText('Gateway description')
+      .should('be.visible')
+      .and('have.text', gateway.description)
+    cy.findDescriptionByLabelText('Gateway description')
+      .should(
+        'contain',
+        'Optional gateway description; can also be used to save notes about the gateway',
+      )
+      .and('be.visible')
+    cy.findByLabelText('Gateway Server address')
+      .should('be.visible')
+      .and('have.attr', 'value', gateway.gateway_server_address)
+    cy.findDescriptionByLabelText('Gateway Server address')
+      .should('contain', 'The address of the Gateway Server to connect to')
+      .and('be.visible')
+    cy.findDescriptionByLabelText('Automatic updates')
+      .should('contain', 'Gateway can be updated automatically')
+      .and('be.visible')
```

Comment:
I think checks like these are overly specific. We shouldn't care about the exact contents of the field descriptions. If anything, we can check whether they are visible, but even that, I wouldn't see it as a crucial matter for testing.